src/IR/IR.hs

```diff
@@ -222,7 +223,7 @@ data Expr t
 
 -- | An alternative in a pattern-match.
 data Alt
-  = AltData DConId [Binder]
+  = AltData DConId [Alt]
   -- ^ @AltData d vs@ matches data constructor @d@, and names dcon members @vs@.
   | AltLit Literal
   -- ^ @AltLit l@ matches against literal @l@, producing expression @e@.
```

Comment:
Can you change `AltDefault` to be `AltBinder` instead? Because "default" no longer makes sense here.